File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/api/ReaderAiUseCases.java
package com.cleverferret.v2.feature.ai.api;

import com.cleverferret.v2.core.common.result.IntegrationResult;
import com.cleverferret.v2.feature.ai.model.AiGenerationResult;

public interface ReaderAiUseCases {
    IntegrationResult<AiGenerationResult> summarizeSelection(String requester, String text);

    IntegrationResult<AiGenerationResult> analyzeCharacters(String requester, String text);

    IntegrationResult<AiGenerationResult> generateMindMap(String requester, String text);
}

File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/api/ReaderAiUseCases.kt
package com.cleverferret.v2.feature.ai.api

import com.cleverferret.v2.core.common.result.IntegrationResult
import com.cleverferret.v2.feature.ai.model.AiGenerationResult

interface ReaderAiUseCases {
    fun summarizeSelection(requester: String, text: String): IntegrationResult<AiGenerationResult>
    fun analyzeCharacters(requester: String, text: String): IntegrationResult<AiGenerationResult>
    fun generateMindMap(requester: String, text: String): IntegrationResult<AiGenerationResult>
}
